### ISSUE TITLE
What is Aidou ?

### ISSUE THREAD

**Issue by DutchPete** [marktext/marktext#1286]
### Question description

 Aidou is part of the General settings in Preferences: what is the effect of enabling it?

**Comment by Jocs** [comment]
![1566210764927](https://user-images.githubusercontent.com/9712830/63258947-c44da900-c2af-11e9-912f-5c67f2b9c9c0.jpg)

The aidou is mainly used to insert some interesting pictures. Chinese people like to add some interesting pictures to chat or blog, so this function is added. When enabled, there will be an Aidou menu item in the Edit menu.

**Comment by fuzzy76** [comment]
This should be in the documentation or at least described in the preferences window.

**Comment by Jocs** [comment]
@fuzzy76 Sorry, we do not have such documents about this, I think we can add to this document https://github.com/marktext/marktext/blob/develop/docs/FAQ.md.

**Comment by pecigonzalo** [comment]
Could we add Giphy? I think it would be a lot more friendly to the western community.

**Comment by generic-user** [comment]
Hi, sorry to intrude in the conversation, but I was asking myself the same question...
Even when disabling it in the options, it's still available in the command menu ...

**Comment by DutchPete** [comment]
@BC: you should ask this question on the Mark Text forum. In any case, I can tell you that Aidou is for Chinese users, kind of like emojis.
-- 
 Securely sent with Tutanota. Get your own encrypted, ad-free mailbox: 
 https://tutanota.com


17 Apr 2020, 10:57 by notifications@github.com:

**Comment by brainchild0** [comment]
Since this service is unfamiliar to many users who are not Chinese, maybe it would be helpful to create configuration sections that are specific to certain languages or regions. Eventually it is likely that other features will be requested that are not useful in Chinese, but are useful in other languages. Then, everyone not using some language feature or regional feature is not distracted by it or confused by it.

**Comment by NewUserHa** [comment]
Who the hell will insert meme in MarkDown by default?
I suggest the function should not be turned on by default.

**Comment by dave-kennedy** [comment]
It might be sufficient to simply relabel the option "Enable Aidou image search" or similar.

I agree it'd be neat if you could configure the service to use, e.g. giphy, but I also don't insert a lot of memes in my docs.

**Comment by vani-lu** [comment]
Even if I am a Chinese user, I could not intuitively understand what the `aidou` setting is.  
I use English language for my working computer, and `aidou` is just completely confusing. And the pop up searchbar when I accidentally press `cmd + /` is even more inexplicable. I might have heard of its Chinese name somewhere, but why should remember it? There ARE people in China who do not care about this function in their life.
If you think it is a function most users need, AT LEAST provide some description in the settings, which explicitly states that `aidou` is a meme gif database for Chinese users. OR add decent user guide into your wiki.

**Comment by fxha** [comment]
We recently added an entry to the FAQ.

@Jocs Do we really need this feature, or would we remove it from Mark Text?

**Comment by brainchild0** [comment]
My view is that this feature is one that eventually might be separated into a plugin, for users who might use it, probably a minority.

Even so, I would suggest emphasizing other opportunities, at the current time,  for improving usability and stability. At some later time, hopefully, it would be possible to consider adding an extensions interface, allowing certain features to be moved outside of the core application. 

In short, I feel that other priorities are higher than removing support for Aidou. As long as  its purpose is explained, I doubt it would harm anyone.

**Comment by Cactusinhand** [comment]
> We recently added an entry to the FAQ.
> 
> @Jocs Do we really need this feature, or would we remove it from Mark Text?

First, this feature does not work properly for a long time. Second, it violates its own philosophy--simple, clean and elegant.

The reason why it remains so far, I guess, may be related to the special sponsors

**Comment by Jocs** [comment]
>Do we really need this feature, or would we remove it from Mark Text?

@fxha Let me add the background of this feature. When I first started writing MarkText, I added this feature. At that time, I added some animated pictures when I wrote the article, so this feature has been retained and has not been removed until now. I also didn't expect this feature to cause so much confusion to everyone. As @brainchild0 said, this feature will be introduced through plug-ins later, and the current main work is on the usability and stability of the editor, so there is no more energy to optimize this feature, at least this feature currently has some Some users are using it, so there is no direct removal.

>First, this feature does not work properly for a long time. Second, it violates its own philosophy--simple, clean and elegant.
>The reason why it remains so far, I guess, may be related to the special sponsors

@Cactusinhand It’s true that this feature is inconsistent with the original intention(philosophy--simple, clean and elegant) of MarkText, but it has nothing to do with special  sponsors. The upload of images to GitHub should be introduced through plugins too, rather than integrated into MarkText. The current introduction of these features is due to user needs and MarkText does not support plug-in functions yet.
